File: C++/BubbleSort.cpp
#include <iostream>
using namespace std;

int main()
{
    int arrayToSort[] = {3,5,1,6,8,2,1,8,1,6,9,0,6,7,4};
	int length = sizeof(arrayToSort) / sizeof(int);

    // Print values
	cout << "- Array before bubble sort: { ";
	for(int i : arrayToSort){
		cout << i << " ";
	}
	cout << "}" << endl;
	// ------------------------------------

    // Bubble sort
    for(int i = 0; i < length; i++){
        for(int j = 0; j < length; j++){
            if(j != 0 && arrayToSort[j] < arrayToSort[j-1]){
                int temp = arrayToSort[j];
                arrayToSort[j] = arrayToSort[j-1];
                arrayToSort[j-1] = temp;
            }
        }
    }
    // ------------------------------------

    // Print result
	cout << "- Array after bubble sort: { ";
	for(int i : arrayToSort){
		cout << i << " ";
	}
	cout << "}";

    return 0;
}

File: Go/BubbleSort.go
package main

import "fmt"

func main() {
	var arrayToSort = []int{3, 5, 1, 6, 8, 2, 1, 8, 1, 6, 9, 0, 6, 7, 4}

	fmt.Print("- Array before bubble sort: ")
	fmt.Println(arrayToSort)

	// Bubble sort
	for i := 0; i < len(arrayToSort); i++ {
		for j := 1; j < len(arrayToSort); j++ {
			if arrayToSort[j] < arrayToSort[j-1] {
				arrayToSort[j], arrayToSort[j-1] = arrayToSort[j-1], arrayToSort[j]
			}
		}
	}

	fmt.Print("- Array after bubble sort: ")
	fmt.Println(arrayToSort)
}
